packages/build/src/barque.ts

```diff
@@ -0,0 +1,190 @@
+import BuildVariant from './build-variant';
+import child_process from 'child_process';
+import gunzip from 'gunzip-maybe';
+import Platform from './platform';
+import fetch from 'node-fetch';
+import tmp from 'tmp-promise';
+import Config from './config';
+import stream from 'stream';
+import fs from 'fs-extra';
+import tar from 'tar-fs';
+import util from 'util';
+import path from 'path';
+
+const pipeline = util.promisify(stream.pipeline);
+const execFile  = util.promisify(child_process.execFile);
+
+const LATEST_CURATOR =
+  'https://s3.amazonaws.com/boxes.10gen.com/build/curator/curator-dist-ubuntu1604-latest.tar.gz';
+
+// make sure everything written in /tmp is cleared if an uncaught exception occurs 
+tmp.setGracefulCleanup()
+
+/**
+ * Distro enum to be used when making a curator call.
+ */
+enum Distro {
+  Ubuntu = 'ubuntu1804',
+  Debian = 'debian10',
+  Redhat = 'rhel80'
+}
+
+/**
+ * Arch enum to be used when making a curator call.
+ *
+ * If we were to target a different arch for these distros, make sure
+ * config/repo-config.yml and packages/build/src/tarball.ts are changed accordingly.
+ */
+enum Arch {
+  Ubuntu = 'amd64',
+  Debian = 'amd64',
+  Redhat = 'x86_64',
+}
```

Comment:
It feels a bit odd to infer the architecture name from the distro name – maybe we should add a `buildArch` key to the config, and then take that into account here as well (in `determineArch()`)? It’s not important for this PR, but maybe the comment above could mention it?

Reply:
the `config` is created based on the variables that evergreen provides for us, so if I were to put `arch` in the `config`, I would be basically doing the same thing as I am here. Which perhaps is not actually bad since we get to keep everything in one place?

Reply:
We could also just make a ticket (or just discuss in design doc) for the release epic to just stop detecting what to do based on which runner we are on and rather pass those stuff as parameters in the scripts / as env var and make that populate the config.

What everyone thinks about that? Is probably not going to change much if we make that change only here.

Reply:
I think that it could be discussed in that next release automation epic!